### ISSUE TITLE
This work is hard to repeat

### ISSUE THREAD

**Issue by Dawn-bin** [mj129/CoANet#4]
I have followed this work for a long time, and the result claims in the paper **can't repeat!!**
I have several qustions about the code and paper:
1. As we know, before the data feed into model, we need do some 'transfoms'. But why do you [padding the ignore region for 0](https://github.com/mj129/CoANet/blob/main/dataloaders/custom_transforms.py#L220) and then using it for loss calculation? And why not mask this region out?
2. About the **Connction Branch**, in your process code, ["create_connction.py"](https://github.com/mj129/CoANet/blob/main/process/create_connection.py), we can see that, it create from the **GroundTruth** using some pixel interval. In the  paper, page 7, it claimed that "**Besides, we devise an upper bound to our proposed method CoANet by using the ground truth of connectivity branch during inference**". It doesn't make  sense! **CoANet-UB is so big just because It get the GroundTruth!**
3. Go on with the **Connction Branch**, It claimed that "The upper bound results indicate that there is still large room for the connectivity branch to improve, and one possibility is to use a larger pixel interval in the future work."(page 7 right) But from the **Ablation Study**, **TABLE V**, page 10, we can see that It is not work by using a larger pixel interval. It is not consistent about your discussion.

Could you answer my qustion?

**Comment by mj129** [comment]
Thanks for your attention to our work.
1. The code of "custom_transforms.py" is based on the pytorch-deeplab-xception (https://github.com/jfzhang95/pytorch-deeplab-xception/blob/master/dataloaders/custom_transforms.py) codebase. If the random scale is smaller than the crop size, the resized image will be padded with 0. 
2. CoANet-UB is the upper bound for our CoANet with the ground truth of the connectivity branch, which is used to verify the predictive ability of the connectivity branch. CoANet in Table I and II is the final results of our model.
3. In Sec. IV-E-2) of our paper, CoANet-d5 represents that our CoANet consists of three CoA modules with d = 1, d = 3, and d = 5, whose results are lower than CoANet. As shown in the visualization results, a bigger pixel interval will bring more background areas. However, it can be concluded from our many experiments that a bigger pixel interval is more effective for extracting roads with large occlusions (i.e. difficult samples), so a larger pixel interval is a possible solution for datasets with many difficult samples.

**Comment by Dawn-bin** [comment]
You not answer the **“Reason”** to do this.
1. Take [Cityscapes](https://github.com/jfzhang95/pytorch-deeplab-xception/blob/master/dataloaders/datasets/cityscapes.py#L84) as a example, the codebase fill with 255. Does there some special consideration?
2. & 3. Your paper make a misleading that connectivity branch has more powerful than it is really does.
